src/App.tsx

```diff
@@ -15,37 +15,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  REVERSE = 'reverse',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [activeSort, setActiveSort] = useState<SortType | ''>('');
```

Comment:
Consider creating one more state for the reverse to get products sorted by length or alphabet and reversed at the same time